src/esm_master/task.py

```diff
@@ -534,3 +550,181 @@ def output_steps(self):
+                if command in subtask.package.command_list.get("get", []):
+                    self.num_get_commands += 1
+                    break
+
+    def get_repo_properties_from_command(self, command):
+        """
+        If the current command is a repo action (e.g. cloning) collects the repo
+        information associated to that command
+
+        Parameters
+        ----------
+        command : str
+            The shell command to be evaluated
+
+        Returns
+        -------
+        repo : dict
+            A dictionary containing the following information:
+            - ``package``: the package object that contains all the package info
+            - ``is_repo_operation``: boolean that indicates whether ``command`` is a
+              repo operation
+            - ``is_last_repo``: boolean that indicates whether this is the last repo
+              operation or not
+        """
+        # Initializes the list of executed repo commands
+        if not hasattr(self, "executed_repo_commands"):
+            self.executed_repo_commands = []
+
+        repo = {"is_repo_operation": False, "is_last_repo": False}
+        get_commands = self.package.command_list.get("get")
+        # This is true for standalone models
+        if self.package.repo and get_commands:
+            if command in get_commands:
+                repo["package"] = self.package
+                repo["is_repo_operation"] = True
+                repo["is_last_repo"] = True
+        # This block here is for coupled setups (loops through each subtask -> model)
+        else:
+            for subtask in self.subtasks:
+                get_commands = subtask.package.command_list.get("get")
+                if command in get_commands:
+                    repo ["package"]= subtask.package
+                    repo["is_repo_operation"] = True
+                    self.executed_repo_commands.append(command)
+                    if self.num_get_commands == len(self.executed_repo_commands):
+                        repo["is_last_repo"] = True
+                    break
+
+        return repo
+
+    def add_repo_error(self, command, repo, error):
+        """
+        If an error occurred during the repo ``command``, then stores information about
+        the error, and the associated package. The information about the error is
+        saved in the attribute ``self.repo_errors`` (if it doesn't exist, it creates
+        it).
+
+        The ``repo_errors`` attribute itself is a ``dict`` with keys being the failed
+        ``command``s, each of them containing the following variables:
+        - ``model``: model name
+        - ``repo``: url of the repo
+        - ``destination``: destination folder
+        - ``error``: "destination path exist" if the destination exist, and the value
+          of the ``error`` parameter if it the destination does not exist
+
+        Note: because the ``error`` is caught from an ``except`` of a ``subprocess``
```

Comment:
Probably not important, but you can use this:
```
Note
----
```
It might show up in a "more beautiful" way in Sphinx.
```suggestion
        Note
        ---- 
        Since the ``error`` is caught from an ``except`` of a ``subprocess``
```